install/ui/src/freeipa/widgets/MigrateScreen.js

```diff
@@ -0,0 +1,215 @@
+                    } else if (result.status === 'invalid-password') {
+                        result.message = this.migration_invalid_password;
+                    }
+                    return result;
+                }
+
+                function error_handler(xhr, text_status, error_thrown) {
+                    topic.publish('rpc-end');
+                    return result;
+                }
+
+                var data = {
+                    username: username,
+                    password: password
+                };
+
+                var request = {
+                    url: config.migration_url,
+                    data: data,
+                    contentType: 'application/x-www-form-urlencoded',
+                    processData: true,
+                    dataType: 'html',
+                    async: false,
+                    type: 'POST',
+                    success: success_handler,
+                    error: error_handler
+                };
+
+                topic.publish('rpc-start');
+                $.ajax(request);
+
+                return result;
+            },
+
+            refresh: function() {
+                this.set('aside', this.migration_info_msg);
+                var val_summary = this.get_widget('validation');
+
+                var u_f = this.fields.get('username');
+                this.get_widget('username').focus_input();
+            },
+
+            constructor: function(spec) {
+                spec = spec || {};
+
+                this.migration_error_msg = text.get(
+                    spec.migration_error_msg ||
+                        '@i18n:migration.migration_error_msg',
+                        this.migration_error_msg);
+
+                this.migration_failure_msg = text.get(
+                    spec.migration_failure_msg ||
+                        '@i18n:migration.migration_failure_msg',
+                        this.migration_failure_msg);
+
+                this.migration_info_msg = text.get(
+                    spec.migration_info_msg ||
+                        '@i18n:migration.migration_info_msg',
+                        this.migration_info_msg);
+
+                this.migration_invalid_password = text.get(
+                    spec.migration_invalid_password ||
+                        '@i18n:migration.migration_invalid_password',
+                        this.migration_invalid_password);
+
+                this.migration_success = text.get(
+                    spec.migration_success ||
+                        '@i18n:migration.migration_success',
+                        this.migration_success);
+
+                this.field_specs = MigrateScreen.field_specs;
+            }
+        });
+
+        MigrateScreen.field_specs = [
+            {
+                $type: 'text',
+                name: 'username',
+                label: text.get('@i18n:login.username', "Username"),
+                placeholder: text.get('@i18n:login.username', "Username"),
```

Comment:
add `required: true,` here and to password field to get required checking for free